r/tests/testthat/test-trackplots.R

```diff
@@ -180,44 +192,54 @@ test_that("trackplot_coverage doesn't crash", {
       atac_reads = as.integer(seq(from=0, to=10, length.out=length(cell_names))),
       cell_type = sample(c("T", "B", "NK"), size = length(cell_names), replace=TRUE)
     )
-    region <- tibble::tibble(
+    # general region
+    region1 <- tibble::tibble(
         chr = "chr15",
         start = 30000000,
         end = 52806155,
     )
-    expect_no_condition(
-        ggplot2::ggplot_build(trackplot_coverage(
-              frags,
-              region=region,
-              groups=frags_metadata$cell_type,
-              cell_read_counts=frags_metadata$atac_reads,
-              bins=500
-            )
-        )
+    # for small motifs
+    region2 <- tibble::tibble(
+        chr = "chr15",
+        start = 31283779,# 31277136,
+        end = 31283870#31322748
     )
-    # checks for only one cluster
     frags_metadata_one_cluster <- frags_metadata[frags_metadata$cell_type == "T",]
+    # checks for only one cluster
     frags_one_cluster <- select_cells(frags,  frags_metadata_one_cluster$id)
-    expect_no_condition(
-        ggplot2::ggplot_build(trackplot_coverage(
-              frags_one_cluster,
-              region=region,
-              groups=frags_metadata_one_cluster$cell_type,
-              cell_read_counts=frags_metadata_one_cluster$atac_reads,
-              bins=500
+    for (region in list(region1, region2)) {
+        expect_no_condition(
+            ggplot2::ggplot_build(trackplot_coverage(
+                  frags,
+                  region=region,
+                  groups=frags_metadata$cell_type,
+                  cell_read_counts=frags_metadata$atac_reads,
+                  bins=500
+                )
             )
         )
-    )
-    # Check return data works
-    expect_is(
-        trackplot_coverage(
-            frags_one_cluster,
-            region=region,
-            groups=frags_metadata_one_cluster$cell_type,
-            cell_read_counts=frags_metadata_one_cluster$atac_reads,
-            bins=500,
-            return_data=TRUE
-        ),
-        "tbl_df"
-    )
+        
+        expect_no_condition(
+            ggplot2::ggplot_build(trackplot_coverage(
+                  frags_one_cluster,
+                  region=region,
+                  groups=frags_metadata_one_cluster$cell_type,
+                  cell_read_counts=frags_metadata_one_cluster$atac_reads,
+                  bins=500
+                )
+            )
+        )
```

Comment:
Running this test on `region2`, I get the following error:
```
Error: Expected `ggplot2::ggplot_build(...)` to run without any conditions.
ℹ Actually got a <simpleWarning> with text:
  NAs introduced by coercion to integer range
```
Can you track down why this is?